src/excmds.ts

```diff
@@ -2519,14 +2519,22 @@ export async function tabclose(...indexes: string[]) {
         }
         return idFromIndex(id)
     }
+    let ids
     if (indexes.length > 0) {
-        const ids = await Promise.all(indexes.map(index => maybeWinTabToTabId(index)))
-        done = browser.tabs.remove(ids)
+        // Request to close multiple tabs
+        ids = await Promise.all(indexes.map(index => maybeWinTabToTabId(index)))
     } else {
-        // Close current tab
-        done = browser.tabs.remove(await activeTabId())
+        // Request to close the current tab
+        ids = [await activeTabId()]
+    }
+    const tabclosepinned = (await config.getAsync("tabclosepinned") === "true")
+    for (let tab_id of ids) {
```

Comment:
We don't need to do this loop if tabclosepinned is true, so I'd move the check up to here

Reply:
You mean, put the loop in a `if (!tabclosepinned) { ... }` ?

Reply:
Done